ocrd_utils/ocrd_utils/str.py

```diff
@@ -79,6 +80,9 @@ def make_file_id(ocrd_file, output_file_grp):
         while ocrd_file.mets.find_files(ID=ret):
             n += 1
             ret = concat_padded(output_file_grp, n)
+    if not REGEX_FILE_ID.fullmatch(ret):
+        ret = ret.replace(':', '_')
+        ret = re.sub(r'^(\d+)', r'id_\1', ret)
```

Comment:
Why not attempt full repair:
```suggestion
        ret = re.sub(r'^([^a-zA-Z_])', r'id_\1', ret)
        ret = re.sub(r'[^\w.-]', r'', ret)
```

This still does not guarantee we are unique, but this is as good as it gets.